File: api/app/graphql/app/physical/types/component.rb
# frozen_string_literal: true

module App
  module Physical
    module Types
      # ComponentType
      class Component < ::Types::BaseObject
        graphql_name 'ComponentTypeT'

        field :id, ID, null: true
        field :name, String, null: false
        # field :chassis_id, ID, null: true
        field :component_type_id, ID, null: true
        field :description, String, null: true
        field :serial_number, String, null: true
      end
    end
  end
end

File: front/app/types/physical/component.ts
import * as v from 'valibot'

// Components
export const componentSchema = v.object({
    id: v.string(),
    name: v.pipe(v.string(), v.nonEmpty()),
    serialNumber: v.nullish(v.string()),
    description: v.nullish(v.string()),
    componentTypeId: v.nullish(v.string()),
})

export const componentFormSchema = v.partial(componentSchema, ['id'])

// Export des types
export type ComponentType = v.InferInput<typeof componentSchema>
// export type ComponentFormType = v.InferInput<typeof componentFormSchema>